numpy/core/src/multiarray/compiled_base.c

```diff
@@ -80,6 +80,358 @@ minmax(const npy_intp *data, npy_intp data_len, npy_intp *mn, npy_intp *mx)
+
+            if (bins[d] <= 0 || (max[d] - min[d]) <= 0)
+            {
+                PyErr_SetString(PyExc_ValueError, "Bin edges must be increasing");
+                goto fail;
+            }
+            norm[d] = bins[d] / (max[d] - min[d]);
+
+            e += bins[d] + 1;
+        }
+    }
+
+    if (weights == Py_None) {
+        /*
+         * weights is None, so output histogram dtype is int
+         */
+        hist = (PyArrayObject *)PyArray_ZEROS(ndim, bins, NPY_INTP, 0);
+        if (hist == NULL) {
+            goto fail;
+        }
+
+        NPY_BEGIN_ALLOW_THREADS;
+        for (i = 0; i < len; i++) {
+            overflow = 0;
+            {
+                double *e = (double *)PyArray_DATA(arr_edges);
+                for (d = 0; d < ndim; d++) {
+                    tx = x[i * ndim + d];
+                    if ((min[d] <= tx) && (tx <= max[d])) {
+                        ix[d] = (tx - min[d]) * norm[d];
+                        /*
+                         * For values that lie exactly on mx we need to subtract one.
+                         * Also, the index computation is not guaranteed to give
+                         * exactly consistent results within ~1 ULP of the bin edges.
+                         * The last bin includes the right edge. The other bins do not.
+                         */
+                        if (ix[d] == bins[d] || (ix[d] != 0 && tx < e[ix[d]])) {
+                            ix[d]--;
+                        }
+                        else if ((ix[d] + 1) != bins[d] && tx >= e[ix[d] + 1]) {
+                            ix[d]++;
+                        }
+                        e += bins[d] + 1;
+                    }
+                    else {
+                        overflow = 1;
+                        break;
+                    }
+                }
+            }
+            if (overflow) {
+                continue;
+            }
+            *((npy_intp *)PyArray_GetPtr(hist, ix)) += 1;
+        }
+        NPY_END_ALLOW_THREADS;
+    }
+    else {
+        /*
+         * weights given, so output histogram dtype is double
+         */
+        arr_w = (PyArrayObject *)PyArray_ContiguousFromAny(weights, NPY_DOUBLE, 1, 1);
+        if (arr_w == NULL) {
+            goto fail;
+        }
+        if (PyArray_DIM(arr_w, 0) != len) {
+            if (!PyErr_Occurred()) {
+                PyErr_SetString(PyExc_ValueError,
+                                "Weights must be same length as array");
+            }
+            goto fail;
+        }
+
+        hist = (PyArrayObject *)PyArray_ZEROS(ndim, bins, NPY_DOUBLE, 0);
+        if (hist == NULL) {
+            goto fail;
+        }
+
+        w = (double *)PyArray_DATA(arr_w);
+
+        NPY_BEGIN_ALLOW_THREADS;
+        for (i = 0; i < len; i++) {
+            overflow = 0;
+            {
+                double *e = (double *)PyArray_DATA(arr_edges);
+                for (d = 0; d < ndim; d++) {
```

Comment:
And this one. Same for `ii` and `tx` too, which could be declared within the loop